libmcount/record.c

```diff
@@ -880,8 +880,11 @@ void record_proc_maps(char *dirname, const char *sess_id,
 			   &start, &end, prot, path) != 4)
 			continue;
 
-		/* skip special mappings like [heap], [stack] */
-		if (path[0] == '[')
+		/*
+		 * skip special mappings like [heap], [vdso] etc.
+		 * but [stack] is needed when getting kernel base address later.
+		 */
+		if (path[0] == '[' && path[1] != 's')
```

Comment:
How about using `strncmp(path, "[stack", 6)`?

Reply:
OK, Got it!